### ISSUE TITLE
More aggressive walk function

### ISSUE THREAD

**Issue by pathunstrom** [ppb/pursuedpybear#665]
The current walk is fine as is, but we need a walk implementation that doesn't care if the child has its own walk method so that the renderer itself can be sure to reach all renderable objects.

We can call the new method `render_walk` for now (unless someone has a better idea) and it should only care that each level has children. It can use the `render_walk` method to get the child and grandchildren, but has to work with the case that `render_walk` isn't implemented.

Part of supporting #577

**Comment by AstraLuma** [comment]
[Current function](https://github.com/ppb/pursuedpybear/blob/cacde17237c75504d8871d11d3fb5d52363c62b8/ppb/gomlib.py#L182-L195) is already pretty aggressive and has minimal dependencies on object methods? What it doesn't do is allow you to prune the walk.

**Comment by pathunstrom** [comment]
It still relies on `walk` doing the right thing. If we want to support "renderer always gets everything" and "can opt out of event dispatch" we need to split this regardless.

**Comment by AstraLuma** [comment]
true, but we have so little design on "can opt out of event dispatch", i wouldn't worry about it.

**Comment by pathunstrom** [comment]
Then how about we change the scope of this to having the current walk step around game objects that don't implement walk?
